core/validation/initial_check.py

```diff
@@ -0,0 +1,150 @@
+"""Initial Pre-transformation Validation
```

Comment:
As a general observation: This code is super-specific and not really reusable at all (other rounds, different funds, different version of forms etc), since lots of the validation values are hard-coded into the functions. 

Since most of the fields this is validating are single fields, being compared against a single value, you could for the most part just re-use something like your "check_form_input" function, for all (or most) of the fields we are checking, and invoke that function with params, field_value_entered and field_value_expected. That way it's much more versatile, when we come to the next round, we can simply pass the same function a new pre-defined selection of expected field values. Does that make sense?

Reply:
@Joel-Pearce this hasn't change hasn't been implemented and I think it's key to what this pre validation pattern should look like. 

check_form_input should look something like this
```
def check_form_input(entered_value: str,  expected_values: str | set[str]) -> vf.PreTransformationFailure:
    if isinstance(expected_values, str):
        expected_values = {expected_values}

    if entered_value not in expected_values:
        return vf.PreTransformationFailure(
                entered_value=entered_value,
                expected_values=expected_values,
            )
        )
```
which we can then just use in a loop on the values extracted from the submission i.e. fund type, form version, place name...

Reply:
I've refactored the code to make pre_transformation_check() round agnositic. Only the extraction function will differ per round.